sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -448,6 +451,23 @@
     }
 
     /**
+     * Use timestamp to set up start offset.
+     * It is only supported by Kafka Client 0.10.1.0 onwards and the message format version
+     * after 0.10.0.
+     *
+     * <p>If you set this, this would ignore {@code ConsumerConfig.AUTO_OFFSET_RESET_CONFIG}
+     * in configuration and any committed offsets.
+     *
+     * <p>There is a failure in the following two cases:
+     * 1.If no message has a timestamp that is greater than or equals to the target time.
+     * 2.If the message format version in a partition is before 0.10.0, i.e. the messages
+     * do not have timestamps.
```

Comment:
Minor tweaks (mainly removing 'you'):

> If you set this, this would ignore {@code ConsumerConfig.AUTO_OFFSET_RESET_CONFIG}
  in configuration and any committed offsets.

Note that this take priority over start offset configuration {@code ConsumerConfig.AUTO_OFFSET_RESET_CONFIG} and any auto committed offsets.

<p> This results in hard failures in either of the following two cases :
  1. If one of more partitions do not contain any messages with timestamp larger than or equal to desired timestamp.
  2. If the message format version in a partition is before 0.10.0, i.e. the messages do not have timestamps.